Src/ILGPU/IR/Construction/Method.Builder.cs

```diff
@@ -324,34 +392,102 @@ public BasicBlock.Builder CreateBasicBlock(Location location, string name)
                 out bool created) =>
                 Context.Declare(declaration, out created);
 
+            /// <summary>
+            /// Computes an updated block collection using the latest terminator
+            /// information.
+            /// </summary>
+            /// <typeparam name="TOrder">The collection order.</typeparam>
+            /// <returns>The newly ordered collection.</returns>
+            public BasicBlockCollection<TOrder, Forwards>
+                ComputeBlockCollection<TOrder>()
+                where TOrder : struct, ITraversalOrder
+            {
+                // Compute new block order
+                var newBlocks = ImmutableArray.CreateBuilder<BasicBlock>(blockCounter);
+                TOrder order = default;
+                order.Traverse<
+                    ImmutableArray<BasicBlock>.Builder,
+                    BasicBlock.TerminatorSuccessorsProvider,
+                    Forwards,
+                    ImmutableArray<BasicBlock>>(
+                    EntryBlock,
+                    newBlocks,
+                    new BasicBlock.TerminatorSuccessorsProvider());
+
+                // Return new block collection
+                return new BasicBlockCollection<TOrder, Forwards>(
+                    EntryBlock,
+                    newBlocks.ToImmutable());
+            }
+
             #endregion
 
             #region IDisposable
```

Comment:
SUGGESTION: The `IDisposable` region includes `UpdateParameters` and `UpdateControlFlow`. These can be moved to the `Methods` region.

Reply:
Makes sense. I thought it might be helpful to move these methods to the `IDisposable` region, since they are only used in the `Dispose` method. However, I fully agree that we should move them to the `Methods` region.

Reply:
Ah, I did not realise that those methods were only used by the `Dispose` method. In that case, it's fine to leave in the `Disposable` region.